packages/frontend/src/components/authenticate/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { Button, Typography } from "@carrot-kpi/ui";
+import React, { useCallback, useEffect, useState } from "react";
+import LogoIcon from "../../icons/logo-icon";
+import { t } from "i18next";
+import { useAccount, useSignMessage } from "wagmi";
+import WalletDisconnected from "../../icons/wallet-disconnected";
+import { useSetPinningProxyJWT } from "../../hooks/useSetPinningProxyJWT";
+
+interface AuthenticateProps {
+    onCancel: () => void;
+}
+
+export const Authenticate = ({ onCancel }: AuthenticateProps) => {
+    const { address } = useAccount();
+    const setPinningProxyJWT = useSetPinningProxyJWT();
+
+    const [loading, setLoading] = useState(false);
+
+    const { data: signedLoginMessage, signMessage } = useSignMessage();
+
+    useEffect(() => {
+        const fetchData = async () => {
+            if (!address || !signedLoginMessage) return;
+            setLoading(true);
+            try {
+                if (!address || !signedLoginMessage) return;
```

Comment:
We can remove this `if` since we're checking for the same condition above.

Reply:
Fixed in 00b83a648f5fa57aab1827856ac5814678ba8fda